src/cautiousrobot/buddy_check.py

```diff
@@ -1,98 +1,89 @@
-import sys
 import pandas as pd
-from cautiousrobot.utils import process_csv
 
 
 class BuddyCheck:
     def __init__(self, buddy_id = None, buddy_col = "md5"):
+        '''
+        Parameters:
+        -----------
+        buddy_id - String [optional]. Name of matching unique identifier column for checksum_df. Matches on both filename and checksum values when provided.
+        buddy_col - String. Column name for checksums in checksum_df (algorithm used for the checksums with sum-buddy). Default: 'md5'.
+        '''
         self.buddy_id = buddy_id
         self.buddy_col = buddy_col
 
 
-    def check_alignment(self, img_df, checksum_df, id_col = "filename", validation_col = "checksum", buddy_id = None, buddy_col = None):
+    def merge_on_checksum(self, source_df, checksum_df, source_validation_col):
         '''
-        Check that all expected images were downloaded.
-        Merges on the filename and checksum columns for both the source file and the checksum file produced by sum-buddy.
-        If buddy_id is not given, merges on just the checksum columns--not recommended if duplicate images are possible.
-        Saves a missing images CSV if there are less than the expected number of matches and prints the number missing.
+        Merge source and checksum DataFrames on only checksum values.
+        '''
+        print(f"merging on checksums only")
+        merged_df = pd.merge(source_df,
+                             checksum_df,
+                             left_on = source_validation_col,
+                             right_on = self.buddy_col,
+                             how = "inner")
+        return merged_df
+    
+    
+    def merge_on_filename_checksum(self, source_df, checksum_df, source_id_col, source_validation_col):
```

Comment:
It might be helpful to name functions and variables related to the ID consistently. E.g. `merge_on_id_and_checksum`, since the default column of `'filename'` may be changed.

Reply:
I do tend to use 'filename' and 'id' somewhat interchangeably in BuddyCheck since the ID is always a filename (the checksum is always for a file whose ID is its filename).